driver-sync/src/test/functional/com/mongodb/client/CrudProseTest.java

```diff
@@ -16,181 +16,411 @@
+    @Test
+    void testBulkWriteCollectsWriteConcernErrorsAcrossBatches() {
+        assumeTrue(serverVersionAtLeast(8, 0));
+        assumeFalse(isServerlessTest());
+        assumeTrue(Runtime.getRuntime().availableProcessors() < 1, "BULK-TODO implement prose test https://github.com/mongodb/specifications/blob/master/source/crud/tests/README.md#5-mongoclientbulkwrite-collects-writeconcernerrors-across-batches");
+    }
+
+    @DisplayName("6. MongoClient.bulkWrite handles individual WriteErrors across batches")
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testBulkWriteHandlesWriteErrorsAcrossBatches(final boolean ordered) {
+        assumeTrue(serverVersionAtLeast(8, 0));
+        assumeFalse(isServerlessTest());
+        assumeTrue(Runtime.getRuntime().availableProcessors() < 1, "BULK-TODO implement prose test https://github.com/mongodb/specifications/blob/master/source/crud/tests/README.md#6-mongoclientbulkwrite-handles-individual-writeerrors-across-batches");
+    }
+
+    @DisplayName("7. MongoClient.bulkWrite handles a cursor requiring a getMore")
+    @Test
+    void testBulkWriteHandlesCursorRequiringGetMore() {
+        assumeTrue(serverVersionAtLeast(8, 0));
+        assumeFalse(isServerlessTest());
+        assumeFalse(isStandalone());
+        assumeTrue(Runtime.getRuntime().availableProcessors() < 1, "BULK-TODO implement batch splitting https://jira.mongodb.org/browse/JAVA-5529");
+        ArrayList<CommandStartedEvent> startedBulkWriteCommandEvents = new ArrayList<>();
+        CommandListener commandListener = new CommandListener() {
+            @Override
+            public void commandStarted(final CommandStartedEvent event) {
+                if (event.getCommandName().equals("bulkWrite")) {
+                    startedBulkWriteCommandEvents.add(event);
+                }
             }
+        };
+        try (MongoClient client = createMongoClient(getMongoClientSettingsBuilder().addCommandListener(commandListener))) {
+            int maxWriteBatchSize = droppedDatabase(client).runCommand(new Document("hello", 1)).getInteger("maxWriteBatchSize");
+            ClientBulkWriteResult result = client.bulkWrite(nCopies(
+                    maxWriteBatchSize + 1,
+                    ClientNamespacedWriteModel.insertOne(namespace(), new Document("a", "b"))));
+            assertEquals(maxWriteBatchSize + 1, result.getInsertedCount());
+            assertEquals(2, startedBulkWriteCommandEvents.size());
+            CommandStartedEvent firstEvent = startedBulkWriteCommandEvents.get(0);
+            CommandStartedEvent secondEvent = startedBulkWriteCommandEvents.get(1);
+            assertEquals(maxWriteBatchSize, firstEvent.getCommand().getArray("ops").size());
+            assertEquals(1, secondEvent.getCommand().getArray("ops").size());
+            assertEquals(firstEvent.getOperationId(), secondEvent.getOperationId());
         }
+    }
```

Comment:
I realized this test can't pass for now only after implementing it. Decided to leave the implementation, though the test is skipped.